source/digits_hits/src/GateRestrictedVolumeMessenger.cc

```diff
@@ -0,0 +1,58 @@
+/** ----------------------
+  Copyright (C): OpenGATE Collaboration
+  This software is distributed under the terms
+  of the GNU Lesser General  Public Licence (LGPL)
+  See LICENSE.md for further details
+  ----------------------*/
+
+
+#include "GateRestrictedVolumeMessenger.hh"
+
+#include "GateRestrictedVolume.hh"
+
+#include "G4UIcmdWithAString.hh"
+
+GateRestrictedVolumeMessenger::GateRestrictedVolumeMessenger(GateRestrictedVolume* itsRestrictedVolume)
+    : GatePulseProcessorMessenger(itsRestrictedVolume)
+{
+  G4String guidance;
+  G4String cmdName;
+  G4String cmdName2;
+
+  cmdName = GetDirectoryName() + "setRestrictedVolume";
+  restrictVolCmd = new G4UIcmdWithAString(cmdName,this);
+  restrictVolCmd->SetGuidance("Set restricted volume for pulse-discrimination");
+
+  cmdName2 = GetDirectoryName() + "setRestrictionMode";
+  restrictModeCmd = new G4UIcmdWithAString(cmdName2,this);
+  restrictModeCmd->SetGuidance("Set restriction mode, two are available: reject accept");
+
+}
+
+GateRestrictedVolumeMessenger::~GateRestrictedVolumeMessenger()
+{
+  delete restrictVolCmd;
+  delete restrictModeCmd;
+}
+
+void GateRestrictedVolumeMessenger::SetNewValue(G4UIcommand* command, G4String newValue)
+{
+  if (command==restrictVolCmd) { RestrictedVolume()->SetRestrictedVolume(newValue); }
+  
+  else if (command==restrictModeCmd) { 
+  
+    if (newValue == "accept" || newValue == "reject")
```

Comment:
Think about define two constant strings kAccept and kReject.
The you can call: `newValue == kAccept || newValue == kReject`
Next to definition of this two variables you should add descriptions what they mean.